html/pages/device/showconfig.inc.php

```diff
@@ -103,10 +103,29 @@
         }
     }
     else if ($config['oxidized']['enabled'] === true && isset($config['oxidized']['url'])) {
+        // Try with hostname as set in librenms first
+        $oxidized_hostname = $device['hostname'];
         // fetch info about the node and then a list of versions for that node
-        $node_info = json_decode(file_get_contents($config['oxidized']['url'].'/node/show/'.$device['hostname'].'?format=json'), true);
+        $node_info = json_decode(file_get_contents($config['oxidized']['url'].'/node/show/'.$oxidized_hostname.'?format=json'), true);
+        
+        // Try other hostname format if Oxidized request failed
+        if (! $node_info) {
+          // Adjust hostname based on whether domain was already in it or not
+          if (strpos($oxidized_hostname, '.') !== false) {
+            // Use short name
+            $oxidized_hostname = strtok($device['hostname'], '.');
+          } else {
+            // Add domain if $config['mydomain'] is set
+            $oxidized_hostname = $config['mydomain'] ? $device['hostname'].'.'.$config['mydomain'] : $oxidized_hostname;
```

Comment:
this would be a little more clear if you used something like this:
```
} elseif($config['mydomain']) {
    $oxidized_hostname = $device['hostname'].'.'.$config['mydomain'];
}
````